events/equeue/equeue_mbed.cpp

```diff
@@ -120,27 +123,28 @@ void equeue_mutex_unlock(equeue_mutex_t *m)
 
 
 // Semaphore operations
-#ifdef MBED_CONF_RTOS_PRESENT
+#ifdef MBED_CONF_RTOS_API_PRESENT
```

Comment:
Can these two compile-paths be consolidated with the baremetal RTOS APIs?

Reply:
It's still theoretically possible to make a "bare metal without RTOS API" build, at the moment - that was one of the issues that didn't really get discussed. (Bullet point in description: "bare-metal: requires rtos-api?")

Or we could actually say that "events" requires "rtos-api" in its JSON, and then drop the other case.

The other case here is a bit lighter in terms of ROM size, but the cost is that you lose the deep sleep latency compensation stuff, which could be problematic. I'm inclined to think that it would be better to drop it and have the only sleep logic in do_sleep.

A possible compile choice that might be worthwhile outside events is lighter-weight versions of the "do_sleep" family. Adding that does increase blinky size - because it now sleeps. I can imagine options for a lighter-weight builds:

* no sleep manager - always just shallow sleep
* Maybe don't even use HAL sleep, just WFI?
* Just spin without timer + sleep/WFI at all - back to the old baremetal `wait_ms` behaviour.